graylog2-web-interface/src/routing/loadAsync.tsx

```diff
@@ -28,7 +30,7 @@ const ErrorComponent = ({ error }: Props) => {
     console.error(error);
   }, [error]);
 
-  return <div>Loading component failed: {error.message}</div>;
+  return <RuntimeErrorPage error={error} componentStack={error.stack} />;
```

Comment:
I noticed that notifications are displayed twice in this case.

<img width="1314" alt="image" src="https://github.com/user-attachments/assets/ec5e4af4-b540-4624-ac16-c4864a4b771f" />

Reply:
We might want to only display the `RuntimeErrorPage` when using `loadAsync` to load a page component.
We also use it in some cases to load components, like the query input and when there is an error it can result in the following UI:

<img width="1264" alt="image" src="https://github.com/user-attachments/assets/4f9bd56a-a9d4-450e-9b3c-52051f91d811" />

Reply:
@linuspahl: Good catch! Instead of rendering the page itself, I changed `loadAsync` to report the error to the store, so the overarching component will pick it up and render it. Does this work better?

Reply:
That works better. Compared with the previous solution it means of course that the whole page (excl. the navigation) is no longer usable, instead of just the area we loaded asynchronously. But that is expected, right?

Reply:
yeah, IMO this way is preferred over a state as shown in the picture above, where parts of the page work and parts do not, e.g. the query input on the search page just not being present. This could lead to all kinds of completely non-deterministic results.